File: lua/tests/test-atomicweight.lua
require("xraylib")
luaunit = require("luaunit")

TestAtomicWeight = {}
	function TestAtomicWeight:test_Fe()
		local weight = xraylib.AtomicWeight(26)
		luaunit.assertAlmostEquals(weight, 55.850)
	end

	function TestAtomicWeight:test_U()
		local weight = xraylib.AtomicWeight(92)
		luaunit.assertAlmostEquals(weight, 238.070)
	end

	function TestAtomicWeight:test_bad_Z()
		luaunit.assertError(xraylib.AtomicWeight, 185)
	end

os.exit(luaunit.LuaUnit.run('-v', '-o', 'tap'))

File: java/tests/TestAtomicWeight.java
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;
import org.junit.jupiter.api.Test;
import com.github.tschoonj.xraylib.Xraylib;

public class TestAtomicWeight {
	@Test
	public void test_Fe() {
		double weight = Xraylib.AtomicWeight(26);
		assertEquals(weight, 55.850, 1E-6);
	}

	@Test
	public void test_U() {
		double weight = Xraylib.AtomicWeight(92);
		assertEquals(weight, 238.070);
	}

	@Test
	public void test_bad_Z() {
		IllegalArgumentException exc = assertThrows(IllegalArgumentException.class, () -> {
			double width = Xraylib.AtomicWeight(185);
		});
		assertEquals(exc.getMessage(), Xraylib.Z_OUT_OF_RANGE);
	}
}
